src/client/chat.js

```diff
@@ -158,7 +158,7 @@ module.exports = function (client, options) {
         message,
         timestamp: options.timestamp,
         salt: options.salt,
-        signature: client.profileKeys ? client.signMessage(message, options.timestamp, options.salt) : Buffer.alloc(0),
+        signature: client.profileKeys ? client.signMessage(options.preview || message, options.timestamp, options.salt) : Buffer.alloc(0),
```

Comment:
Is this correct? The signature check on the server will fail with this.

Reply:
Then the server signature check is incorrect aswell. Vanilla server accepts this and will kick with "Invalid or missing signature" if we use message instead